File: Phase2/src/main/java/InputScannerView.java
package main.java;

import java.util.Scanner;
import java.util.Set;

public class InputScannerView {
    final String backString = "--back";

    public InputScannerView(InvertedIndex index) {
        Scanner scanner = new Scanner(System.in);
        InputScanner inputScanner = new InputScanner(index);
        while (true) {
            String input = scanner.nextLine();
            if (input.equals(backString)) break;
            showResult(inputScanner.getOrder(input));
        }
        scanner.close();

    }

    private void showResult(Set<String> answer) {
        for (String s : answer) System.out.println(s);
    }

}

File: WebApplication/InputScannerView.cs
using System;
using System.Collections.Generic;

namespace ConsoleApp1
{
    public class InputScannerView
    {
        private const string BackString = "--back";

        public InputScannerView()
        {
            
        }

        private static void ShowResult(IEnumerable<string> answer)
        {
            foreach (var s in answer)
            {
                Console.Out.WriteLine(s);
            }
        }

        public IEnumerable<string> start(InvertedIndex index, string input)
        {
            var inputScanner = new InputScanner(index);
            Console.WriteLine(input+")))))))))");
            return inputScanner.GetOrder(input);
        }
    }
}